src/Tribe/iCal.php

```diff
@@ -237,77 +316,282 @@ private function get_month_view_events() {
-	 * @param int|null $post If you want the ical file for a single event
-	 * @param boolean $echo Whether the content should be echoed or returned
+	 * @since TBD
 	 */
-	public function generate_ical_feed( $post = null, $echo = true ) {
-		$tec         = Tribe__Events__Main::instance();
-		$events      = '';
-		$blogHome    = get_bloginfo( 'url' );
-		$blogName    = get_bloginfo( 'name' );
+	protected function set_headers() {
+		header( 'HTTP/1.0 200 OK', true, 200 );
+		header( 'Content-type: text/calendar; charset=UTF-8' );
+		header(
+			'Content-Disposition: attachment; filename="' . $this->get_file_name() . '"'
+		);
+	}
 
-		if ( $post ) {
-			$events_posts = is_array( $post ) ? $post : array( $post );
-		} elseif ( tribe_is_month() ) {
-			$events_posts = self::get_month_view_events();
-		} elseif ( tribe_is_organizer() ) {
-			$events_posts = $this->get_events_list( array(
-				'organizer'    => get_the_ID(),
-				'eventDisplay' => 'list',
-			) );
-		} elseif ( tribe_is_venue() ) {
-			$events_posts = $this->get_events_list( array(
-				'venue'        => get_the_ID(),
-				'eventDisplay' => tribe_get_request_var( 'tribe_event_display', 'list' ),
-			) );
-		} else {
+	/**
+	 * Get the file name of the *.ics file
+	 *
+	 * @since TBD
+	 *
+	 * @return mixed The calendar name
+	 */
+	protected function get_file_name() {
+		$event_ids = wp_list_pluck( $this->events, 'ID' );
+		$site = sanitize_title( get_bloginfo( 'name' ) );
+		$hash = substr( md5( implode( $event_ids ) ), 0, 11 );
+		$filename = sprintf( '%s-%s.ics', $site, $hash );
 
-			if ( ! $wp_query = tribe_get_global_query_object() ) {
-				return;
+		/**
+		 * Modifies the filename provided in the Content-Disposition header for iCal feeds.
+		 *
+		 * @var string       $filename
+		 * @var WP_Post|null $post
+		 */
+		return apply_filters( 'tribe_events_ical_feed_filename', $filename, $this->post );
+	}
+
+	/**
+	 * Get the full content of the *.ics file.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	protected function get_content() {
+		return sprintf(
+			"%s%s%s%s",
+			$this->get_start(),
+			$this->get_timezones( $this->events ),
+			$this->get_body( $this->events ),
+			$this->get_end()
+		);
+	}
+
+	/**
+	 * Get the start of the .ics File
+	 *
+	 * @since TBD
+	 *
+	 * @return mixed
+	 */
+	protected function get_start() {
+		$blog_home    = get_bloginfo( 'url' );
+		$blog_name    = get_bloginfo( 'name' );
+
+		$content  = "BEGIN:VCALENDAR\r\n";
+		$content .= "VERSION:2.0\r\n";
+		$content .= 'PRODID:-//' . $blog_name . ' - ECPv' . Tribe__Events__Main::VERSION . "//NONSGML v1.0//EN\r\n";
+		$content .= "CALSCALE:GREGORIAN\r\n";
+		$content .= "METHOD:PUBLISH\r\n";
+
+		/**
+		 * Allows for customizing the value of the generated iCal file's "X-WR-CALNAME:" property.
+		 *
+		 * @param string $blog_name The value to use for "X-WR-CALNAME"; defaults to value of get_bloginfo( 'name' ).
+		 */
+		$x_wr_calname = apply_filters( 'tribe_ical_feed_calname', $blog_name );
```

Comment:
Docblock with a since

Reply:
This one existed before this release so the `@since` will not be accurate.